### PR TITLE
Allow other partition tables and filesystems

### PR DESCRIPTION
**Pull request by tstat**

### CODE REVIEW

File: rpi/default.nix [hunk 1]
```diff
@@ -304,14 +320,8 @@ in
...
-          # This is ugly and fragile, but the sdImage image has an msdos
-          # table, so the partition table id is a 1-indexed hex
-          # number. So, we drop the hex prefix and stick on a "02" to
-          # refer to the root partition.
-          "root=PARTUUID=${lib.strings.removePrefix "0x" config.sdImage.firmwarePartitionID}-02"
-          "rootfstype=ext4"
-          "fsck.repair=yes"
-          "rootwait"
+          "console=tty1"
+          "console=serial0,115200n8"
```
Comment (MinerSebas) [inline rpi/default.nix:324]: This comment wasn't moved.
```suggestion
          # https://github.com/raspberrypi/firmware/issues/1539#issuecomment-784498108
          "console=tty1"
          "console=serial0,115200n8"
```

File: sd-image/default.nix [hunk 2]
```diff
@@ -9,7 +9,16 @@
     boot.consoleLogLevel = lib.mkDefault 7;
 
     # https://github.com/raspberrypi/firmware/issues/1539#issuecomment-784498108
```
Comment (MinerSebas) [inline sd-image/default.nix:11]:
```suggestion
```

**Review by MinerSebas — COMMENTED** [review]
Does this also fix setting `boot.initrd.systemd.enable = true;`?
I broke my installation when using it.
My uneducated guess is, that this is the culprit:
https://github.com/NixOS/nixpkgs/blob/0f1c50858f16ba4e9743abd607cba7c66e87510a/nixos/modules/system/boot/systemd/initrd.nix#L390-L391


Also the readme isn't updated yet with this change.

**Comment by tstat** [comment]
thanks for the review @MinerSebas!

**Comment by Nebucatnetzer** [comment]
I'm a bit confused by this pull request.
I can see that we can customize the firmware partition but the root partition as far as I can tell or is there a way to override the `rootfsImage` variable?
And what about the `fsType` further down?

https://github.com/nix-community/raspberry-pi-nix/pull/79/files#diff-4ecb3d0364b87606a85cf915fd2580e2367912b08f0b14c1dad6eca051a540e1R25

**Pull request merged**